Modules/Quotas/src/Quotas.Application/IntegrationEvents/Incoming/TierOfIdentityChanged/TierOfIdentityChangedIntegrationEventHandler.cs

```diff
@@ -0,0 +1,31 @@
+﻿using Backbone.Modules.Quotas.Application.Infrastructure.Persistence.Repository;
+using Backbone.Modules.Quotas.Application.Metrics;
+using Enmeshed.BuildingBlocks.Application.Abstractions.Infrastructure.EventBus;
+
+namespace Backbone.Modules.Quotas.Application.IntegrationEvents.Incoming.TierOfIdentityChanged;
+public class TierOfIdentityChangedIntegrationEventHandler : IIntegrationEventHandler<TierOfIdentityChangedIntegrationEvent>
+{
+    private readonly IIdentitiesRepository _identitiesRepository;
+    private readonly ITiersRepository _tiersRepository;
+    private readonly IMetricStatusesService _metricStatusesService;
+
+    public TierOfIdentityChangedIntegrationEventHandler(IIdentitiesRepository identitiesRepository, ITiersRepository tiersRepository, IMetricStatusesService metricStatusesService)
+    {
+        _identitiesRepository = identitiesRepository;
+        _tiersRepository = tiersRepository;
+        _metricStatusesService = metricStatusesService;
+    }
+
+    public async Task Handle(TierOfIdentityChangedIntegrationEvent @event)
+    {
+        var identity = await _identitiesRepository.Find(@event.IdentityAddress, CancellationToken.None, track: true);
+        var newTier = await _tiersRepository.Find(@event.NewTier, CancellationToken.None, track: true);
+        identity.ChangeTier(newTier);
+        await _identitiesRepository.Update(identity, CancellationToken.None);
+        await _metricStatusesService.RecalculateMetricStatuses(
+            new List<string>() { identity.Address },
+            identity.MetricStatuses.Select(x => x.MetricKey).ToList(),
```

Comment:
What if the new Tier has a Quota on a Metric the old Tier didn't have a Metric on? I think in that case there won't be an existing `MetricStatus`.
So you should rather don't pass anything, which should result in all metric statuses being recalculated.